### PR TITLE
Adds a suicide vest

### PR DESCRIPTION
**Pull request by karmaisblackandbluepilled**
self explanatory
I might've taken the code from goon but I can't remember if I did, probably not thoguh
## Changelog
:cl:
add: Added the craftable suicide vest.
/:cl:

### CODE REVIEW

File: code/modules/clothing/suits/armor.dm [hunk 1]
```diff
@@ -409,28 +409,28 @@
...
 
-/obj/item/clothing/suit/armor/suicide/attackby(var/obj/item/grenade/G, var/mob/user)
+/obj/item/clothing/suit/armor/suicide/attackby(obj/item/grenade/G, user)
 	if(istype(G))
 		if(attached_grenade.len >= 4)
 			to_chat(user, "<span class='warning'>All the grenade slots are filled, تؤخر!</span>")
-		else if(user.transferItemToLoc(G,src))
-			user.visible_message("<span class='warning'>\The [user] attaches \a [G] to \the [src]!</span>", "<span class='notice'>You attach \the [G] to \the [src].</span>")
+		else if(usr.transferItemToLoc(G,src))
+			usr.visible_message("<span class='warning'>\The [user] attaches \a [G] to \the [src]!</span>", "<span class='notice'>You attach \the [G] to \the [src].</span>")
```
Comment (TheChosenEvilOne) [inline code/modules/clothing/suits/armor.dm:425]: uhhhh, user, not usr.

File: code/modules/clothing/suits/armor.dm [hunk 2]
```diff
@@ -409,28 +409,28 @@
...
 			attached_grenade.Add(G)
 			G.forceMove(src)
 	else
 		return ..()
 
-/obj/item/clothing/suit/armor/suicide/attack_self(var/mob/user)
+/obj/item/clothing/suit/armor/suicide/attack_self(/user)
 	if(attached_grenade.len)
-		user.visible_message("<span class='warning'>\The [user] empties out \the [src]!</span>", "<span class='notice'>You empty out \the [src].</span>")
+		usr.visible_message("<span class='warning'>\The [usr] empties out \the [src]!</span>", "<span class='notice'>You empty out \the [src].</span>")
```
Comment (TheChosenEvilOne) [inline code/modules/clothing/suits/armor.dm:433]: ditto

**Review by TheChosenEvilOne — CHANGES_REQUESTED** [review]

**Comment by karmaisblackandbluepilled** [comment]
changed

**Comment by EdgeLordExe** [comment]
you better remember where you got this code from, because if you got it from goon you will have to recode it yourself, as their code license is incompatible with ours

**Comment by karmaisblackandbluepilled** [comment]
yeah I think I recoded it myself because last I remember goon adds a check for every single grenade in the code, it doesn't check for the grenade parent type

**Comment by karmaisblackandbluepilled** [comment]
image
ok yeah I found the code from where I pr'd it to hippie. since I didn't add any new sprites, it's all mine babyyy

**Comment by EdgeLordExe** [comment]
if sprites are from goon you still have to put them inside of goon/ folder

**Comment by EdgeLordExe** [comment]
nevermind no new sprites

**Comment by karmaisblackandbluepilled** [comment]
edge learn to read challenge(impossible)
I didn't add any new sprites. These were already in the codebase.

**Pull request merged**